app/src/main/java/fr/free/nrw/commons/profile/leaderboard/DataSourceFactory.java

```diff
@@ -0,0 +1,77 @@
+package fr.free.nrw.commons.profile.leaderboard;
+
+import androidx.lifecycle.MutableLiveData;
+import androidx.paging.DataSource;
+import fr.free.nrw.commons.auth.SessionManager;
+import fr.free.nrw.commons.mwapi.OkHttpJsonApiClient;
+import io.reactivex.disposables.CompositeDisposable;
+
+public class DataSourceFactory extends DataSource.Factory<Integer, LeaderboardList> {
+
+    private MutableLiveData<DataSourceClass> liveData;
+    private OkHttpJsonApiClient okHttpJsonApiClient;
+    private CompositeDisposable compositeDisposable;
+    private SessionManager sessionManager;
+    private String duration;
+    private String category;
+    private int limit;
+    private int offset;
+
+    public String getDuration() {
```

Comment:
I know it is boring, but some simple Javadoc would be better for these methods too.